feature/policy_forwarding/otg_tests/mpls_gre_ipv4_decap_test/README.md

```diff
@@ -0,0 +1,383 @@
+
+#### Ten or more subinterfaces (customer) with different VLAN-IDs
+
+* Two or more VLANs with IPV4 link local address only, /29 address
+
+* Two or more VLANs with IPV4 global /30 address
+
+* Two or more VLANs with IPV6 address /125 only
+
+* Four or more VLANs with IPV4 and IPV6 address
+
+#### L3 Address resolution
+
+* Local proxy ARP for IPV4 (Required for traffic forwarding by DUT to any destinations within same subnet shared between DUT and Port-Channel1)
+
+* Local proxy for IPV6 or support Secondary address for IPV6 allowing resolution of same subnet IPV6 addresses corresponding to remote Cloud endpoints
+
+* Disable Neighbor discovery router advertisement, duplicate address detection
+
+#### MTU Configuration
+* One or more VLANs with MTU 9080 (including L2 header)
+
+#### LLDP must be disabled
+
+### Port-Channel 2 and Port-Channel 3 configuration
+
+* IPV4 and IPV6 addresses
+
+* MTU (default 9216)
+
+* LACP Member link configuration
+
+* Lag id
+
+* LACP (default: period short)
+
+* Carrier-delay (default up:3000 down:150)
+
+* Statistics load interval (default:30 seconds)
+
+### Routing
+
+* MPLSoGRE decapsulation prefix range must be configurable on the device.  MPLSoGRE traffic within one or more prefix ranges must be processed by the device.
+
+* Static mapping of MPLS label to an egress nexthop must be configurable. Egress nexthop is based on the MPLS label/ static LSP.
+
+* MPLS label for a single egress VLAN interface must be unique for decapsulated traffic:
+    * IPV4 traffic
+    * IPV6 traffic
+    * Multicast traffic
+
+* ECMP (Member links in Port Channel1) based on:
+    * inner IP packet header  AND/OR
+    * MPLS label, Outer IP packet header 
+
+* Inner packet TTL and DSCP must be preserved during decap of traffic from MPLSoGRE to IP traffic
+
+### MPLS Label 
+
+* Entire Label block must be reallocated for static MPLS
+
+* Labels from start/end/mid ranges must be usable and configured corresponding to MPLSoGRE decapsulation
+
+### Multicast
+
+* Multicast traffic must be decapsulated and sent out with L2 header based on the multicast payload address.
+
+
+## PF-1.5.1: Verify MPLSoGRE decapsulate action for IPv4 and IPV6 payload
+Generate traffic on ATE Ports 3,4,5,6 having:
+* Outer source address: random combination of 1000+ IPV4 source addresses
+* Outer destination address: Traffic must fall within the configured IPV4 unicast decap prefix range for MPLSoGRE traffic on the device
+* MPLS Labels: Configure streams that map to every egress interface by having associated IPV4/IPV6/Multicast static MPLS labels in the MPLSoGRE header
+* Inner payload: 
+  * Both IPV4 and IPV6 unicast payloads, with random source address, destination address, TCP/UDP source port and destination ports
+* Use 64, 128, 256, 512, 1024.. MTU bytes frame size
+
+Verify:
+
+* Egress routes are programmed and LACP bundles are up without any errors, chassis alarms or exception logs
```

Comment:
How should all these things verified? You have to specify exactly how to verify this in custom instructions here in this test, or test helpers that have already been defined to do these things.